lib/set_game/game_server.ex

```diff
@@ -1,19 +1,53 @@
 defmodule SetGame.GameServer do
-  use GenServer
-
+  use GenServer, start: {__MODULE__, :start_link, []}, restart: :transient
   alias SetGame.Board
   alias SetGame.Player
 
   defmodule State do
-    defstruct board: Board.new(), players: [], state: :players_joining
+    defstruct board: Board.new(),
+              name: "",
+              players: [],
+              state: :players_joining
   end
 
   # Client
 
-  def start_link(_arg \\ nil) do
-    GenServer.start_link(__MODULE__, %State{})
+  @spec start_link(any()) :: :ignore | {:error, any} | {:ok, pid}
+  def start_link({:via, _, {_, name}} = via_tuple_name \\ via_tuple(find_available_name())) do
+    GenServer.start_link(__MODULE__, fresh_state(name), name: via_tuple_name)
   end
 
+  defp timeout(), do: Application.get_env(:set_game, :timeout)
```

Comment:
rather than having this be a function your could make it a compile-time variable:

```elixir
@timeout Application.get_env(:set_game, :timeout)
```

Then just use `@timeout` instead of `timeout()` in the app code. The compiler will in-line the value

Reply:
The tradeoff is that you cannot then adjust the value at runtime

Reply:
I guess that limitation does not matter for this use case.

Thank you both!

Reply:
Oh, another problem. The config is a `runtime.exs`, so if it is a module attribute, it returns `nil`. If it is not runtime, the application might not be able to change it at all. I'll keep it as it is.